packages/edit-site/src/components/global-styles/shadows-edit-panel.js

```diff
@@ -73,12 +73,25 @@ const presetShadowMenuItems = [
 
 export default function ShadowsEditPanel() {
 	const {
+		goBack,
 		params: { category, slug },
-		goTo,
 	} = useNavigator();
 	const [ shadows, setShadows ] = useGlobalSetting(
 		`shadow.presets.${ category }`
 	);
+	const currentShadow = shadows?.find( ( shadow ) => shadow.slug === slug );
+
+	useEffect( () => {
+		// If the shadow being edited doesn't exist anymore in the global styles setting, navigate back
+		// to prevent the user from editing a non-existent shadow entry.
+		// This can happen, for example:
+		// - when the user deletes the shadow
+		// - when the user resets the styles while editing a custom shadow
+		if ( ! currentShadow ) {
+			goBack();
```

Comment:
[As with the font size presets](https://github.com/WordPress/gutenberg/pull/65791#discussion_r1783157173), we may need to use `goTo()`, otherwise when we press the back button in the presets panel, it will take us to the top screen instead of the Shadows panel:

https://github.com/user-attachments/assets/faa6f2a0-178e-4529-a7f0-542beab73718

Reply:
Good catch. I don't think that `goTo` is the solution, though, because internally `goBack` and `goTo` are essentially the same.

`goBack` is the correct choice, and we should use it instead of `goTo` when we need to navigate back, ideally.

In this case, I managed to fix this by checking also for the `slug`